src/proxy-ui-api/frontend/src/views/AddSubsystem/AddSubsystem.vue

```diff
@@ -1,83 +1,317 @@
+        :selectableClients="selectableClients"
+        @cancel="showSelectClient = false"
+        @save="saveSelectedClient"
+      />
+    </div>
+  </div>
 </template>
 
 <script lang="ts">
 import Vue from 'vue';
-import { RouteName } from '@/global';
+import { mapGetters } from 'vuex';
+import HelpIcon from '@/components/ui/HelpIcon.vue';
+import LargeButton from '@/components/ui/LargeButton.vue';
+import SubViewTitle from '@/components/ui/SubViewTitle.vue';
+import SubsystemDetailsPage from './SubsystemDetailsPage.vue';
+import SelectClientDialog from '@/views/AddClient/SelectClientDialog.vue';
+import FormLabel from '@/components/ui/FormLabel.vue';
+import { Key, Token } from '@/types';
+import { RouteName, UsageTypes } from '@/global';
+import { Client } from '@/types';
+import { ValidationProvider, ValidationObserver } from 'vee-validate';
+
+import * as api from '@/util/api';
 
 export default Vue.extend({
-  data: () => ({
-    member: {
-      name: 'Kuikka',
-      class: 'Org',
-      code: '111',
+  components: {
+    FormLabel,
+    LargeButton,
+    ValidationObserver,
+    ValidationProvider,
+    SelectClientDialog,
+    SubViewTitle,
+  },
+  props: {
+    clientId: {
+      type: String,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      currentStep: 1,
+
+      disableDone: false,
+      certificationService: undefined,
+      filteredServiceList: [],
+      showSelectClient: false as boolean,
+      registerChecked: true,
+    };
+  },
+  computed: {
+    ...mapGetters(['client', 'selectableClients', 'reservedClients']),
+    memberClass: {
+      get(): string {
+        return this.$store.getters.memberClass;
+      },
+      set(value: string) {
+        this.$store.commit('setMemberClass', value);
+      },
+    },
+
+    memberCode: {
+      get(): string {
+        return this.$store.getters.memberCode;
+      },
+      set(value: string) {
+        this.$store.commit('setMemberCode', value);
+      },
+    },
+
+    subsystemCode: {
+      get(): string {
+        return this.$store.getters.subsystemCode;
+      },
+      set(value: string) {
+        this.$store.commit('setSubsystemCode', value);
+      },
+    },
+
+    selectedMember: {
+      get(): Client {
+        return this.$store.getters.selectedMember;
+      },
+      set(value: Client) {
+        this.$store.commit('setMember', value);
+      },
+    },
+
+    duplicateClient(): boolean {
+      if (!this.memberClass || !this.memberCode || !this.subsystemCode) {
+        return false;
+      }
+
+      if (
+        this.reservedClients.some((e: Client) => {
+          if (e.member_class.toLowerCase() !== this.memberClass.toLowerCase()) {
+            return false;
+          }
+
+          if (e.member_code.toLowerCase() !== this.memberCode.toLowerCase()) {
+            return false;
+          }
+
+          if (e.subsystem_code !== this.subsystemCode) {
+            return false;
+          }
+          return true;
+        })
```

Comment:
Wrap in to a separate function for better readability

Reply:
Fixed. It's using a shared function now.